modules/mauth-signer-akka-http/src/main/scala/com/mdsol/mauth/MAuthRequestSigner.scala

```diff
@@ -7,35 +7,42 @@ import java.util.UUID
 import com.mdsol.mauth.util.{CurrentEpochTimeProvider, EpochTimeProvider, MAuthKeysHelper}
 
 import scala.util.{Failure, Success, Try}
+import scala.collection.JavaConverters
 
 /**
   * Library agnostic representation of the data required for a request signing
   *
   * @param httpMethod The HTTP verb of this API call
   * @param uri        The URI of the API call , (host name and port not included)
   * @param body       The body of the request in string form
+  * @param parameters The query parameters of the request in string form
+  *
   */
-case class UnsignedRequest(httpMethod: String = "GET", uri: URI, body: Option[String] = None, headers: Map[String, String] = Map.empty)
+case class UnsignedRequest(httpMethod: String = "GET", uri: URI, body: Option[String] = None, parameters: String = "", headers: Map[String, String] = Map.empty)
 
 /**
-  * Library agnostic representation of a signed request, including header data
+  * Library agnostic representation of a signed request, including header data for V1 or V2
+  *
+  * @note it includes V2 headers only if V2 only is enabled, otherwise it includes the both V1 and V2 headers
   *
-  * @param req        The original request that was used to create this object
-  * @param authHeader The Auth header information
-  * @param timeHeader The Time header information
+  * @param req          The original request that was used to create this object
+  * @param authHeaders  The Auth header information
   */
-case class SignedRequest(req: UnsignedRequest, authHeader: String, timeHeader: String)
+case class SignedRequest(req: UnsignedRequest, authHeaders: Map[String, String])
```

Comment:
This breaks binary compatibility

Reply:
Updated SignedRequest https://github.com/mdsol/mauth-jvm-clients/pull/92/commits/f09b5f748be5b11e107ec3296906f97e2f3eadb5

Reply:
It still isn't binary compatible unfortunately. Let's address this in another PR.